### PR TITLE
Relicense to MIT or Apache-2.0; add license and copyright to FAQ.

### LINKED ISSUE #84: Re-license as Apache/MIT?
Cargo is dual-licensed so I assume dual licensing will make it easier to merge.  The sooner, the better to get buy-in from all copyright holders (ie contributors)

**Note::** I am speaking from a position of ignorance on licensing

### PR DESCRIPTION
**Pull request by obi1kenobi**
Resolves #84.

### CODE REVIEW

**Review by epage — APPROVED** [review]
As one of the contributors, I approve of my copyrighted material being re-licensed under these terms

**Comment by obi1kenobi** [comment]
I added the license and copyright text structure I've used on past projects that have had lawyers review and approve them, so I think structurally it's good enough.

@epage if you'd like to be a co-copyright holder, happy to have you there. You've certainly done a large amount of work here and it wouldn't be at all out of place. I just know that some companies can be a bit touchy about copyrights and their employees, so I didn't want to do it unilaterally. I'm happy to add you if you'd like, and you're also welcome to just push to this branch and add yourself if you'd prefer that (e.g. if legal name != display name for example).

In any case, I won't merge this without a ✅ from you, since we need signoff from all contributors — so far that's just you and me.

File: LICENSE-APACHE
Comment (epage) [inline LICENSE-APACHE:189]: Technically, you aren't the sole copyright holder.

I've also heard that copyright date ranges are superfluous.

Comment (epage) [reply LICENSE-APACHE:189]: >  if you'd like to be a co-copyright holder, happy to have you there. Yo

Unless there is explicit copyright re-assignment, all contributors hold the copyright on their contribution.

Comment (obi1kenobi) [reply LICENSE-APACHE:189]: Please do insert your name there and in the `README.md`, or let me know if you'd prefer `Ed Page` or something else and I can add it.

I once worked with a lawyer who felt the date range was important, so unless it hurts something I'm a bit inclined to keep it.

Comment (epage) [reply LICENSE-APACHE:189]: > Please do insert your name there and in the README.md, or let me know if you'd prefer Ed Page or something else and I can add it.

Are you going to have every contributor edit the file to include their name?  I've not heard of a project doing that.

> I once worked with a lawyer who felt the date range was important, so unless it hurts something I'm a bit inclined to keep it.

I've heard from laywers the opposite :)

Comment (obi1kenobi) [reply LICENSE-APACHE:189]: I just took a peek at the cargo repo, and it seems that the `LICENSE-APACHE` file there just left the `Copyright [yyyy] [name of copyright owner]` line unchanged. We could do the same?

As another option, we could add text to the effect of "Copyright: contributors to this repo, as determined by git commit history" or some such thing.

I'm not a lawyer nor can I play one on TV, so I don't know what the right call is here. Since this code is hopefully going into cargo which is your domain, I'm happy to let you pick what you think is best.

Comment (epage) [reply LICENSE-APACHE:189]: Deferring to git is a nice middle ground for now.

Technically, explicit copyright notices aren't needed due to the Berne Convention (e.g. some people put copyright headers in their code).   The main reason people do any of this is to be explicit about what is implicit and if any part gets pulled out of context, at least some will be left behind.

Comment (obi1kenobi) [reply LICENSE-APACHE]: Let me know what you think of the current wording. I don't have strong feelings about it, so if you have a preference for other wording, let me know or even just push to this branch directly — I don't mind.

Comment (epage) [reply LICENSE-APACHE]: Good enough for me.

**Review by epage — APPROVED** [review]

**Pull request merged**
